### PR TITLE
Handle the app removal event from Slack.

### PR DESCRIPTION
**Pull request by thepug**
Slack has a 'Remove App' button on each app page.  This button is meant for a slack user to remove any added application to their workspace.  An event is generated `tokens_revoked` from slack when this occurs.  This change to auth subscribes the auth service to events in the slack router and when the `tokens_revoked` event occurs it probably removes the slack information.

Deploy to staging with https://github.com/belucid/open-company-infrastructure/pull/65

To test:
- Run the slack router service and follow the instructions to get slack events:
https://github.com/open-company/open-company-slack-router#slack-proxy-ngrok
- Subscribe an SQS queue to the slack events.  Set this queue in your environment ie 
`export AWS_SQS_SLACK_ROUTER_AUTH_QUEUE=carrot-local-slack-router-auth-dev-nathan`

- Signup with slack with local dev.
- Visit the slack app page https://opencompanyhq.slack.com/apps/A5SGHA79P-carrot-local-development?page=1  
- [x] After clicking remove app and logging in again are you asked to add the bot?

### CODE REVIEW

**Comment by thepug** [comment]
[Listen for bot and user removal from Slack and do the needful on our side](https://trello.com/c/EV8e9Vw9/1267-listen-for-bot-and-user-removal-from-slack-and-do-the-needful-on-our-side)

**Comment by thepug** [comment]
Need to address this issue on staging: https://sentry.io/opencompany/oc-staging-auth/issues/561354037/

**Comment by thepug** [comment]
resolved and ready for review https://github.com/open-company/open-company-auth/pull/44#issuecomment-390776319

File: src/oc/auth/app.clj [hunk 1]
```diff
@@ -57,6 +58,7 @@
     "Database: " c/db-name "\n"
     "Database pool: " c/db-pool-size "\n"
     "AWS SQS email queue: " c/aws-sqs-email-queue "\n"
+    "AWS SQS email queue: " c/aws-sqs-slack-router-auth-queue "\n"
```
Comment (bago2k4) [inline src/oc/auth/app.clj]: @thepug  You print email queue 2 times, the second should be slack queue or something else, right?

File: src/oc/auth/async/slack_router.clj [hunk 2]
```diff
@@ -0,0 +1,124 @@
...
+      (when (= type "event_callback")
+        (let [event-type (:type event)]
+          (when (= event-type "tokens_revoked")
+            (let [slack-team-id (:team_id body)]
+              (pool/with-pool [conn db-pool]
+                (let [team (:team-id (first
+                                      (team-res/list-teams-by-index
+                                         conn
+                                         :slack-orgs
+                                         slack-team-id)))]
```
Comment (bago2k4) [inline src/oc/auth/async/slack_router.clj]: @thepug i think you need a for here. You need to get all the carrot teams that have that slack org associated, they may be more then one if i got it right. For each of them remove the associated slack org and then remove the slack org itself from our db.
WYT?

File: src/oc/auth/async/slack_router.clj [hunk 3]
```diff
@@ -0,0 +1,124 @@
...
+    'event_time' 1494281750}
+  "
+  [db-pool body]
+  (let [type (:type body)
+        token (:token body)
+        event (:event body)
+        channel (:channel event)
+        thread (:thread_ts event)]
+    ;; Token check
+    (if-not (= token config/slack-verification-token)
```
Comment (bago2k4) [inline src/oc/auth/async/slack_router.clj:50]: @thepug i don't have any `slack-verification-token` set for localhost, where can i find it?

**Review by bago2k4 — CHANGES_REQUESTED** [review]
@thepug one issues, a couple to discuss

File: src/oc/auth/async/slack_router.clj (hunk 3 above)
Comment (thepug) [reply src/oc/auth/async/slack_router.clj:50]: `export OPEN_COMPANY_SLACK_VERIFICATION_TOKEN="xxxx"` is in my environment.

File: src/oc/auth/async/slack_router.clj (hunk 2 above)
Comment (thepug) [reply src/oc/auth/async/slack_router.clj]: Yeah, I added a `doseq`.

File: src/oc/auth/app.clj (hunk 1 above)
Comment (thepug) [reply src/oc/auth/app.clj]: cut n paste for the win

**Review by bago2k4 — APPROVED** [review]

**Pull request merged**